sdk-tests/src/test/java/io/dapr/it/spring/data/DaprKeyValueRepositoryIT.java

```diff
@@ -0,0 +1,142 @@
+/*
+ * Copyright 2024 The Dapr Authors
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package io.dapr.it.spring.data;
+
+import io.dapr.testcontainers.Component;
+import io.dapr.testcontainers.DaprContainer;
+import io.dapr.testcontainers.DaprLogLevel;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Tag;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.test.context.ContextConfiguration;
+import org.springframework.test.context.DynamicPropertyRegistry;
+import org.springframework.test.context.DynamicPropertySource;
+import org.springframework.test.context.junit.jupiter.SpringExtension;
+import org.testcontainers.containers.Network;
+import org.testcontainers.containers.PostgreSQLContainer;
+import org.testcontainers.junit.jupiter.Container;
+import org.testcontainers.junit.jupiter.Testcontainers;
+
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+import static io.dapr.it.spring.data.DaprSpringDataConstants.BINDING_NAME;
+import static io.dapr.it.spring.data.DaprSpringDataConstants.STATE_STORE_NAME;
+import static org.junit.jupiter.api.Assertions.*;
+
+/**
+ * Integration tests for {@link DaprKeyValueRepositoryIT}.
+ */
+
+@ExtendWith(SpringExtension.class)
+@ContextConfiguration(classes = TestDaprSpringDataConfiguration.class)
+@Testcontainers
+@Tag("testcontainers")
+public class DaprKeyValueRepositoryIT {
+  private static final String CONNECTION_STRING =
+      "host=postgres-repository user=postgres password=password port=5432 connect_timeout=10 database=dapr_db_repository";
+  private static final Map<String, String> STATE_STORE_PROPERTIES = createStateStoreProperties();
+
+  private static final Map<String, String> BINDING_PROPERTIES = Collections.singletonMap("connectionString", CONNECTION_STRING);
+
+  private static final Network DAPR_NETWORK = Network.newNetwork();
+
+  @Container
+  private static final PostgreSQLContainer<?> POSTGRE_SQL_CONTAINER = new PostgreSQLContainer<>("postgres:16-alpine")
+      .withNetworkAliases("postgres-repository")
+      .withDatabaseName("dapr_db_repository")
+      .withUsername("postgres")
+      .withPassword("password")
+      .withExposedPorts(5432)
```

Comment:
This is why it is having problems. The integration tests must use ports detected at runtime and not a fixed one.

Reply:
withExposePort is not needed because PostgreSQLContainer already have it. This is not a fixed port exposed in the host, it declared the port exposed from the container and the host port will be a random one.